testcases/OpTestHMIHandling.py

```diff
@@ -0,0 +1,349 @@
+    #  @param i_ffdcDir Optional param to indicate where to write FFDC
+    #
+    # "Only required for inband tests" else Default = None
+    # @param i_lparIP The IP address of the LPAR
+    # @param i_lparuser The userid to log into the LPAR
+    # @param i_lparPasswd The password of the userid to log into the LPAR with
+    #
+    def __init__(self, i_bmcIP, i_bmcUser, i_bmcPasswd,
+                 i_bmcUserIpmi, i_bmcPasswdIpmi, i_ffdcDir=None, i_lparip=None,
+                 i_lparuser=None, i_lparPasswd=None):
+        self.cv_BMC = OpTestBMC(i_bmcIP, i_bmcUser, i_bmcPasswd, i_ffdcDir)
+        self.cv_IPMI = OpTestIPMI(i_bmcIP, i_bmcUserIpmi, i_bmcPasswdIpmi,
+                                  i_ffdcDir)
+        self.cv_LPAR = OpTestLpar(i_lparip, i_lparuser, i_lparPasswd)
+        self.cv_SYSTEM = OpTestSystem(i_bmcIP, i_bmcUser, i_bmcPasswd,
+                 i_bmcUserIpmi, i_bmcPasswdIpmi, i_ffdcDir, i_lparip,
+                 i_lparuser, i_lparPasswd)
+        self.util = OpTestUtil()
+
+    ##
+    # @brief This is a common function for all the hmi test cases. This will be executed before
+    #        any test case starts. Basically this provides below requirements.
+    #        1. Validates all required lpar commands
+    #        2. It will clone skiboot source repository
+    #        3. Compile the necessary tools xscom-utils and gard utility to test HMI.
+    #        4. Get the list Of Chips and cores in the form of dictionary.
+    #           Ex: {0: [2,3,5,6,a,c,d,f], 1: [1,2,4,5,a,b,d,e]}
+    #        5. In-order to inject HMI errors on cpu's, cpu should be running,
+    #           so disabling the cpu sleep states 1 and 2.
+    #
+    def test_init(self):
+        # Get OS level
+        self.cv_LPAR.lpar_get_OS_Level()
+
+        # Check whether git and gcc commands are available on lpar
+        self.cv_LPAR.lpar_check_command("git")
+        self.cv_LPAR.lpar_check_command("gcc")
+
+        l_dir = "/tmp/skiboot"
+        self.cv_LPAR.lpar_clone_skiboot_source(l_dir)
+        self.cv_LPAR.lpar_compile_xscom_utilities(l_dir)
+        self.cv_LPAR.lpar_compile_gard_utility(l_dir)
+
+        l_res = self.cv_LPAR.lpar_run_command("cd %s/external/xscom-utils/; ./getscom -l" % l_dir)
+        l_cmd = "cat /sys/firmware/opal/msglog | grep -i CHIP"
+        l_res = self.cv_LPAR.lpar_run_command(l_cmd)
+        l_dic = {}
+        self.l_dic = {}
+        l_res = l_res.splitlines()
+        for line in l_res:
+            matchObj = re.search("Chip (\d) Core ([a-z0-9])", line)
+            if matchObj:
+                if l_dic.has_key(matchObj.group(1)):
+                    (l_dic[matchObj.group(1)]).append(matchObj.group(2))
+                else:
+                    l_dic[matchObj.group(1)] = list(matchObj.group(2))
+        self.l_dic = l_dic
+        self.l_dir = l_dir
+        self.cv_LPAR.lpar_run_command("cat /sys/devices/system/cpu/cpu*/cpuidle/state2/disable")
```

Comment:
some comments might be helpful to understand what you are trying to achieve with these commands